app/views.py

```diff
@@ -385,40 +477,49 @@ def edit_post(groupID,threadID,postID):
         db.session.commit()
         return redirect(url_for('forum_post', groupID=groupID,threadID=threadID))
 
-    return render_template('posts.html', title=' | Forum', form=form, editpost=post)
+    return render_template('posts.html', title=' | Forum', postForm=form, editpost=post)
 
 # Routes to delete class
-@app.route('/class/delete', methods=['GET', 'POST'])
+@app.route('/class/<int:groupID>/delete', methods=['GET','POST'])
 @login_required
-def delete_class():
+def delete_class(groupID):
      form = DeleteClassForm()
      if form.validate_on_submit():
-         code = Group.query.filter_by(classCode = form.code.data).first()
-         return redirect(url_for('delete_class_confirm'))
+         group = Group.query.filter_by(id=groupID, classCode = form.title.data).first_or_404()
+         remove_group(group)
+         flash('Class deleted')
+         return redirect(url_for('dashboard'))
      return render_template('deleteclass.html', title=' | Deactivate Account', form=form)
 
-#def delete_class_confirm():
-#    I will leave the backend to you xD
-#    return render_template('dashboard.html')
 
 # Routes to edit participants list
-@app.route("/class/participants", methods=['GET', 'POST'])
+@app.route("/class/<int:groupID>/participants")
 @login_required
 def edit_participants(groupID):
+    if not current_user.check_educator():
+        return render_template('error404.html'), 404
```

Comment:
Refer to the previously raised issue regarding errors

Reply:
```suggestion
        return render_template('error403.html'), 403
```